src/utils/proxySet.ts

```diff
@@ -0,0 +1,96 @@
+import { proxy } from 'valtio'
+
+// properies that we don't want to expose to the end-user
+interface InternalProxySet<T> extends Set<T> {
```

Comment:
```suggestion
type InternalProxySet<T> = Set<T> & {
```
Just preferring type alias when we don't need type overloading.